### ISSUE TITLE
how to disable built-in plugins for a channel

### ISSUE THREAD

**Issue by daomengRen** [sopel-irc/sopel#2398]
### Description

I have set the following parameters in the configuration file of the cfg
```
[#channel]
disable_plugins = admin,adminchannel,announce,bugzilla,calc,choose,clock,coretasks,countdown,currency,dice,emoticons,find,find_updates,help,invite,ip,isup,lmgtfy,meetbot,ping,pronouns,py,rand,reddit,reload,remind,safety,search,seen,tell,tld,translate,unicode_info,units,uptime,url,version,wikipedia,wiktionary,xkcd
```
But when sopel starts, the screen doesn't print the join channel message, I think it has an error, but I don't know how to fix it

### Reproduction steps

Add the following configuration to the configuration file
```
[#channel]
disable_plugins = admin,adminchannel,announce,bugzilla,calc,choose,clock,coretasks,countdown,currency,dice,emoticons,find,find_updates,help,invite,ip,isup,lmgtfy,meetbot,ping,pronouns,py,rand,reddit,reload,remind,safety,search,seen,tell,tld,translate,unicode_info,units,uptime,url,version,wikipedia,wiktionary,xkcd
```

### Expected behavior

All built in plugins should be disabled

### Sopel version

7.1.9

### Installation method

`pip install`

### Python version

3.7.6

### Operating system

Debain 9

**Comment by dgw** [comment]
1. This should have been a [Discussion](https://github.com/sopel-irc/sopel/discussions/new/choose); it is not a bug report.
2. Did you read the [documentation](https://sopel.chat/docs/configuration.html#plugins)? The first configuration example shows how to disable plugins using the `[core]` section's `exclude` option.
  - See also: The `-a`/`--allow-only` flag to [`sopel-plugins enable`](https://sopel.chat/docs/cli.html#sopel-plugins-enable)

**Comment by dgw** [comment]
I admit, I rushed in before realizing that you gave example configuration for [per-channel options](https://sopel.chat/usage/per-channel-configuration/). That ought to teach me exactly how wise it is to check tickets at almost 3am. (I'm sure I won't learn that lesson…) I've flagged my previous comment as "off-topic", hiding it. Sorry about that.

The _real_ problem here is that you included `coretasks` in the list of plugins to disable. `coretasks` is required for _anything_ to work; it's what handles most of the IRC protocol messages, including JOINs.

We're not likely to change the code so it's impossible to exclude `coretasks` here, certainly not in the 7.1.x release series, because all of this per-channel stuff is long overdue for a rewrite; we plan to throw most of it away and start over.
